deployment/ccip/changeset/internal/deploy_home_chain.go

```diff
@@ -185,10 +186,61 @@ func BuildSetOCR3ConfigArgs(
 	return offrampOCR3Configs, nil
 }
 
+type MultiOCR3BaseOCRConfigArgsSolana struct {
+	ConfigDigest                   [32]byte
+	OcrPluginType                  uint8
+	F                              uint8
+	IsSignatureVerificationEnabled bool
+	Signers                        [][20]uint8
+	Transmitters                   []solana.PublicKey
+}
+
+// BuildSetOCR3ConfigArgsSolana builds OCR3 config for Solana chains
+func BuildSetOCR3ConfigArgsSolana(
+	donID uint32,
+	ccipHome *ccip_home.CCIPHome,
+	destSelector uint64,
+) ([]MultiOCR3BaseOCRConfigArgsSolana, error) {
+	ocr3Configs := make([]MultiOCR3BaseOCRConfigArgsSolana, 0)
+	for _, pluginType := range []types.PluginType{types.PluginTypeCCIPCommit, types.PluginTypeCCIPExec} {
+		ocrConfig, err2 := ccipHome.GetAllConfigs(&bind.CallOpts{
+			Context: context.Background(),
+		}, donID, uint8(pluginType))
+		if err2 != nil {
+			return nil, err2
+		}
+
+		// we expect only an active config and no candidate config.
+		if ocrConfig.ActiveConfig.ConfigDigest == [32]byte{} || ocrConfig.CandidateConfig.ConfigDigest != [32]byte{} {
+			return nil, fmt.Errorf("invalid OCR3 config state, expected active config and no candidate config, donID: %d", donID)
+		}
+
+		activeConfig := ocrConfig.ActiveConfig
+		var signerAddresses [][20]uint8
+		var transmitterAddresses []solana.PublicKey
+		for _, node := range activeConfig.Config.Nodes {
+			var signer [20]uint8
+			copy(signer[:], node.SignerKey[:20])
```

Comment:
Haven't read the rest of the code yet but shouldn't `node.SignerKey` be exactly 20 bytes long? Could be simplified to


```suggestion
            // can assert len(node.SignerKey) == 20, error otherwise
			copy(signer[:], node.SignerKey[:])
```

Reply:
good catch